neural_modelling/src/neuron/c_main.c

```diff
@@ -95,8 +96,8 @@ int32_t rewiring_period = 0;
 //! Flag representing whether rewiring is enabled
 bool rewiring = false;
 
-// FOR DEBUGGING!
-uint32_t count_rewires = 0;
+//! DEBGGING: for counting the number of rewiring attempts
```

Comment:
if this is going into prov data. its no longer for debugging. its now for prov.

Reply:
Technically speaking this value is counting the number of attempts, whereas what goes into the provenance is the number of successful rewires.  But I have updated the comment anyway as there's no real need to say it's for debugging either.